tempo_embeddings/embeddings/weaviate_database.py

```diff
@@ -0,0 +1,278 @@
+from weaviate.classes.query import Filter
+from weaviate.classes.query import MetadataQuery
+from weaviate.util import generate_uuid5
+from ..text.corpus import Corpus
+from ..text.highlighting import Highlighting
+from ..text.passage import Passage
+from .vector_database import VectorDatabaseManagerWrapper
+
+
+Collection = TypeVar("Collection")
+logger = logging.getLogger(__name__)
+
+class WeaviateDatabaseManager(VectorDatabaseManagerWrapper):
+    """A Weaviate Database can create N collections and will always use THE SAME embedder function and tokenizer for all collections.
+    To create collections with different embedders one needs to create separate databases
+    """
+
+    def __init__(
+        self,
+        db_path: str = "weaviate_default_db",
+        embedder_name: str = None,
+        embedder_config: dict[str, Any] = None,
+        batch_size: int = 8,
+    ):
+        super().__init__(batch_size)
+        self.db_path = db_path
+        self.embedder_name = embedder_name
+        self.embedder_config = embedder_config or {"type": "default"}
+        self.tokenizer = AutoTokenizer.from_pretrained(embedder_name) if embedder_name else None
+        self.model = None
+        self.client = None
+        self.weaviate_headers = {}
+        
+        if self.embedder_config.get("type") == "hf":
+            self.embedding_function = wvc.config.Configure.Vectorizer.text2vec_huggingface(model=self.embedder_name)
+            self.weaviate_headers = {"X-HuggingFace-Api-Key": self.embedder_config["api_key"]}
+        elif self.embedder_config.get("type") == "custom_model" or self.embedder_config.get("type") == "default":
+            try:
+                self.model = self.embedder_config["model"]
+                self.model.batch_size = self.batch_size
+                self.embedding_function = wvc.config.Configure.Vectorizer.none()
+            except KeyError:
+                logger.error("If the type is 'custom_model' or 'default' you should pass the model object under Key 'model'")
+        else:
+            raise ValueError(f"Malformed embedder_config {self.embedder_config}. Check that 'type', 'api_key' and 'model' keys are properly populated.")
+
+        if os.path.exists(self.db_path):
+            self._load_config()
+        else:
+            os.makedirs(self.db_path)
+            self.config = {
+                "embedder_name": self.embedder_name,
+                "embedder_type": self.embedder_config["type"],
+                "existing_collections": [],
+            }
+        
+        
+    def connect(self):
+        with weaviate.connect_to_local() as client:
+            # Do something with the client
+            logger.info(type(client))
+            logger.info("Weaviate Server Is Up: %s", client.is_ready())
+            return client.is_ready()
+
+
+    def _save_config(self):
+        if self.config:
+            with open(f"{self.db_path}/config.json", "w", encoding="utf-8") as f:
```

Comment:
This silently overwrites an existing `config.json`. Not sure if that is intended behaviour here. If not, use `x` mode to raise an exception if the file already exists:
```suggestion
            with open(f"{self.db_path}/config.json", "x", encoding="utf-8") as f:
```